### PR TITLE
fix(nuxt): treeshake `<DevOnly>` with webpack

### LINKED ISSUE #21012: `DevOnly` is not treeshaken with webpack
### Environment

Nuxi 3.5.0                                                               
                                                                           
RootDir: J:/GIT/nuxt   

### Reproduction

https://github.com/huang-julien/nuxt/tree/dev_only_treeshake_webpack

### Describe the bug

DevOnly is not being treeshaken with webpack. I think it has the same origin as https://github.com/nuxt/nuxt/issues/20833

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
fix https://github.com/nuxt/nuxt/issues/21012

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/core/plugins/dev-only.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const DevOnlyPlugin = createUnplugin((options: DevOnlyPluginOptions) => {
       const { type } = parseQuery(search)
 
       // vue files
-      if (pathname.endsWith('.vue') && (type === 'template' || !search)) {
+      if (pathname.endsWith('.vue') && (meta.framework === 'webpack' ? true : type === 'template' || !search)) {
```
Comment (ineshbose) [inline packages/nuxt/src/core/plugins/dev-only.ts:23]:
```suggestion
      if (pathname.endsWith('.vue') && (meta.framework === 'webpack' || type === 'template' || !search)) {
```

Comment (huang-julien) [reply packages/nuxt/src/core/plugins/dev-only.ts]: :+1:

Comment (ineshbose) [reply packages/nuxt/src/core/plugins/dev-only.ts]: Although my preference goes to this if we need to include more (unlikely though)

```suggestion
      if (pathname.endsWith('.vue') && (['webpack'].includes(meta.framework) || type === 'template' || !search)) {
```

**Review by ineshbose — APPROVED** [review]

File: packages/nuxt/src/core/plugins/dev-only.ts (hunk 1 above)
Comment (huang-julien) [reply packages/nuxt/src/core/plugins/dev-only.ts]: we only support webpack and vite for now, i think a PR for rspack is on the way, maybe should we wait for it and change the implementation ?

**Review by danielroe — APPROVED** [review]

**Pull request merged**